proseco/core.py

```diff
@@ -996,6 +1003,28 @@ def null_logger(*args, **kwargs):
 
             return null_logger
 
+    def set_agasc_supp_version(self):
+        """
+        Get the versions of the agasc supplement used for the star data for catalog
+        determination and save in the appropriate MetaAttributes.
+        """
+
+        # It seems incorrect to save the supplement versions that were used to
+        # update stars if the supplement was not actually enabled/used.
+        # This block copies the logic from agasc.update_from_supplement
+        # (which perhaps should be generalized in the interface).
+        supplement_enabled_env = os.environ.get(agasc.SUPPLEMENT_ENABLED_ENV,
+                                                agasc.agasc.SUPPLEMENT_ENABLED_DEFAULT)
+        if supplement_enabled_env not in ('True', 'False'):
+            raise ValueError(
+                f'{agasc.SUPPLEMENT_ENABLED_ENV} env var must be either "True" or "False" '
+                f'got {supplement_enabled_env}')
```

Comment:
I didn't see a sensible way around this which suggests to me that it might be helpful to add something to the agasc module to handle the logic.  But I realize that I'm the only one interested in this.

Reply:
I agree that nothing out of agasc should be doing these kind of manipulations to interpret the environment variables.

Also, I am assuming it would not be possible to fool this determination by using the `set_supplement_enabled` context manager. 

It also bugs me that even though this is a boolean flag, it does a string comparison. I mean, it bugs me to read:
```
if supplement_enabled_env == 'True':
```
instead of 
```
if supplement_enabled_env:
```

Reply:
I don't know if I've thought of the ways somebody might fool this determination using `set_supplement_enabled`.  I think `agasc` probably needs a `get_supplement_enabled` state checker to isolate scope to that module.  *But* something like this could work as a `proseco` update without more changes in the `agasc` module...

Reply:
With regard to it being a string comparision, it sounds to me like the string comparison isn't the issue... you are just bothered that environment variables don't have even-weak types?

Reply:
I would first interpret the env variables to produce a bool, and then use the bool in the if statement. In this case it is the same, because the if statement follows right away, so it is just a matter of style.

If you merge this change, and then if you ever modify how the env variables are interpreted inside agasc, you might get an inconsistency. I know I would forget about this. This is the main issue (and you original comment). That's why it might be a good idea to do it right.

Reply:
Right, so I agree that I thought this proseco code was awkward, but I figured I'd write it out as a bit of a use case so we could define the best changes in `agasc`.  I would figure we'd pull out the block in update supplement https://github.com/sot/agasc/blob/dddfab67a67b783c97af358fd14e4fbbf1fcd8cd/agasc/agasc.py#L503 into a new method and export it.

Reply:
Yes. It is a bit more than that:
- First thing is to move the code in [agasc.py#L504](https://github.com/sot/agasc/blob/master/agasc/agasc.py#L504) into its own function.
- Then one needs to add this function to `agasc. __all__`
- and remove `'SUPPLEMENT_ENABLED_ENV'`. The point here is that SUPPLEMENT_ENABLED_ENV is not the right thing to expose, because the right thing is a combination of SUPPLEMENT_ENABLED_ENV and SUPPLEMENT_ENABLED_DEFAULT.

Since this was explicitly named in `agasc.__all__`, then the question is if it is in use in other places already.

Reply:
Yes, that's exactly what I meant in the previous comment.

Reply:
> and remove 'SUPPLEMENT_ENABLED_ENV'

You can't remove this env var definition because this is the programmatic pathway to disabling use of the supplement for an entire module.  See e.g.: 
https://github.com/sot/proseco/blob/20dcd52b2d8f57f80752fd00d8ce06656f65d9f3/proseco/tests/test_catalog.py#L28

The API here is strict, namely the value of that environment variable (which is by definition a string) can have one of two values: `"False"` or `"True"`. Anything else is an exception and one gets a clear message of what went wrong and how to fix it. 

With that API defined it really is not a bad thing that string comparisons are used given the very simple logic that is contained in a few lines. Sure, we could import `ast` and do an `ast.literal_eval()` of the variable, but that hasn't really made things easier to understand IMO.

Moving the test logic (the line 504 stuff) into a separate function that is also exposed in the public API is a good idea.

Reply:
I was thinking two things:

1) would remove SUPPLEMENT_ENABLED_ENV from proseco because that would be encapsulated in a new public method.

2) I was thinking that the programmatic pathway to disability use of the supplement in Python could just use another public method that encapsulated it, but maybe I have not thought that through.